kubeadm: add section on how to use the "generate-csr" command (#43871)
kubernetes/website: content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md

needs, you should let kubeadm manage certificates instead.
{{< /caution >}}

> ### Set up a signer {#setup-a-signer}

Reviewer: (nit)
This will break existing hyperlinks
```suggestion
### Set up a signer
```

        -in "${FILE}.csr" -out "${FILE}.crt"
done

# Process front-proxy CSRs
echo "* Processing ./pki/front-proxy-client.csr ..."
openssl x509 -req -days "${DAYS}" -CA ./pki/front-proxy-ca.crt -CAkey ./pki/front-proxy-ca.key -CAcreateserial \
    -in ./pki/front-proxy-client.csr -out ./pki/front-proxy-client.crt
```

### Embedding certificates in kubeconfig files

Repeat this step for all nodes that have CSR files.

Write the following script in the `/etc/kubernetes` directory, navigate to the directory
and execute the script. The script will take the `.crt` files that were signed for
kubeconfig files from CSRs in the previous step and will embed them in the kubeconfig files.

```bash
#!/bin/bash

CLUSTER=kubernetes
find ./ -name "*.conf" | while read -r FILE;
do
    echo "* Processing ${FILE} ..."
    KUBECONFIG="${FILE}" kubectl config set-cluster "${CLUSTER}" --certificate-authority ./pki/ca.crt --embed-certs
    USER=$(KUBECONFIG="${FILE}" kubectl config view -o jsonpath='{.users[0].name}')
    KUBECONFIG="${FILE}" kubectl config set-credentials "${USER}" --client-certificate "${FILE}.crt" --embed-certs
done
```

> ### Performing cleanup

Reviewer: (nit)
There might be more than one clean up section one day.
```suggestion
### Performing cleanup {#cleanup-issue}
```

Author: named it `{#performing-cleanup}`

Author: could be named post-csr-cleanup too.

> Ideally: move the tasks into separate pages within /tasks/

after this, the certs page would not have any major additions, i would think.

Perform this step on all nodes that have CSR files.

Write the following script in the `/etc/kubernetes` directory, navigate to the directory
and execute the script.

```bash
#!/bin/bash

# Cleanup CSR files
rm -f ./*.csr ./pki/*.csr ./pki/etcd/*.csr # Clean all CSR files

# Cleanup CRT files that were already embedded in kubeconfig files
rm -f ./*.crt
```

Optionally, move `.srl` files to the next node to be processed.

Optionally, if using external CA remove the `/etc/kubernetes/pki/ca.key` file,
as explained in the [External CA node](#external-ca-mode) section.

### kubeadm node initialization

> Once CSR files have been signed and required certificates are in place on nodes,
> the kubeadm commands `kubeadm init` and `kubeadm join` can be used to create
> a Kubernetes cluster from these nodes. During `init` and `join` kubeadm will
> use existing certificates, keys and kubeconfig files that it finds in the
> `/etc/kubernetes` tree.

Reviewer: (nit)
```suggestion
Once CSR files have been signed and required certificates are in place on the hosts
you want to use as nodes, you can use the commands `kubeadm init` and `kubeadm join`
to create a Kubernetes cluster from these nodes. During `init` and `join`, the kubeadm
tool uses existing certificates, encryption keys and kubeconfig files that it finds in the
`/etc/kubernetes` tree on the host's local filesystem.
```
